R/get_osw_query.R

```diff
@@ -254,8 +316,33 @@ getPrecursorsQuery <- function(runType = "DIA_Proteomics", level = "Peptide")
       WHERE PRECURSOR.DECOY = 0
       ORDER BY peptide_id, transition_group_id, transition_id;"
     }
-  } else if (runType == "DIA_Metabolomics")
-  {
+  } else if (runType == "DIA_IPF"){
+    ## TODO: Not as clear on PROTEIN level, for now we only do peptide level
+      query <- "SELECT DISTINCT PRECURSOR.ID AS transition_group_id,
+      TRANSITION_PRECURSOR_MAPPING.TRANSITION_ID AS transition_id,
+      PEPTIDE.ID AS peptide_id,
+      PEPTIDE.MODIFIED_SEQUENCE AS sequence,
+      PRECURSOR.CHARGE AS charge,
+      PRECURSOR.GROUP_LABEL AS group_label
+      --TRANSITION.DETECTING AS detecting,
+	    --TRANSITION.IDENTIFYING AS identifying
+      FROM PRECURSOR
+      INNER JOIN TRANSITION_PRECURSOR_MAPPING ON TRANSITION_PRECURSOR_MAPPING.PRECURSOR_ID = PRECURSOR.ID
+      INNER JOIN TRANSITION ON TRANSITION.ID = TRANSITION_PRECURSOR_MAPPING.TRANSITION_ID
+      INNER JOIN PRECURSOR_PEPTIDE_MAPPING ON PRECURSOR_PEPTIDE_MAPPING.PRECURSOR_ID = PRECURSOR.ID
+      INNER JOIN PEPTIDE ON PRECURSOR_PEPTIDE_MAPPING.PEPTIDE_ID = PEPTIDE.ID
+      INNER JOIN (
+      SELECT PEPTIDE_ID
+      FROM SCORE_PEPTIDE
+      WHERE SCORE_PEPTIDE.CONTEXT = $CONTEXT AND SCORE_PEPTIDE.QVALUE < $FDR
+      ) AS SCORE_PEPTIDE ON SCORE_PEPTIDE.PEPTIDE_ID = PEPTIDE.ID
+      WHERE PRECURSOR.DECOY = 0
+      AND (
+      TRANSITION.DETECTING=TRUE
+      OR TRANSITION.IDENTIFYING=$USE_IDENTIFYING --- #identifying
```

Comment:
How does IPF calculate intensity? How many and which transitions does it uses?

Reply:
So only DETECTING transitions will be used to create a new feature if it is missing